### ISSUE TITLE
Scylla repeatedly fails to start when `rpc_address` contains a typo

### ISSUE THREAD

**Issue by argenet** [scylladb/scylla#4476]
*Inqstallation details*
Scylla version (or git commit hash): 3.0.5-0.20190414.403f66eca
Cluster size: 1 node
OS (RHEL/CentOS/Ubuntu/AWS AMI): Ubuntu 16.04

I have accidentally mistyped the `rpc_address` to contain a backtick:
```
rpc_address: 127.0.0.1`
```

(pay attention to the backtick after 127.0.0.1 - it is important for reproduction).

Then I tried to start Scylla with `systemctl start scylla-server` and it failed multiple times repeatedly.
The `journalctl _COMM=scylla` shows:
```
May 03 09:11:50 vladimir-ubuntu scylla[34940]: Scylla version 3.0.5-0.20190414.403f66eca starting ...
May 03 09:11:50 vladimir-ubuntu scylla[34940]:  [shard 0] seastar - Exiting on unhandled exception: std::system_error (error C-Ares:4,
May 03 09:11:50 vladimir-ubuntu scylla[34940]: scylla: ./seastar/core/sharded.hh:377: seastar::sharded<T>::~sharded() [with Service = g
May 03 09:11:50 vladimir-ubuntu scylla[34940]: Aborting on shard 0.
May 03 09:11:50 vladimir-ubuntu scylla[34940]: Backtrace:
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   0x00000000006a92f2
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   0x00000000005c640c
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   0x00000000005c66b5
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   0x00000000005c6763
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   /lib/x86_64-linux-gnu/libpthread.so.0+0x000000000001138f
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   /lib/x86_64-linux-gnu/libc.so.6+0x0000000000035427
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   /lib/x86_64-linux-gnu/libc.so.6+0x0000000000037029
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   /lib/x86_64-linux-gnu/libc.so.6+0x000000000002dbd6
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   /lib/x86_64-linux-gnu/libc.so.6+0x000000000002dc81
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   0x0000000000f37662
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   0x0000000000534c6a
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   /lib/x86_64-linux-gnu/libc.so.6+0x000000000002082f
May 03 09:11:50 vladimir-ubuntu scylla[34940]:   0x00000000005a1a98
May 03 09:11:59 vladimir-ubuntu scylla[34966]: Scylla version 3.0.5-0.20190414.403f66eca starting ...
May 03 09:11:59 vladimir-ubuntu scylla[34966]:  [shard 0] seastar - Exiting on unhandled exception: std::system_error (error C-Ares:4,
May 03 09:11:59 vladimir-ubuntu scylla[34966]: scylla: ./seastar/core/sharded.hh:377: seastar::sharded<T>::~sharded() [with Service = g
May 03 09:11:59 vladimir-ubuntu scylla[34966]: Aborting on shard 0.
May 03 09:11:59 vladimir-ubuntu scylla[34966]: Backtrace:
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   0x00000000006a92f2
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   0x00000000005c640c
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   0x00000000005c66b5
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   0x00000000005c6763
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   /lib/x86_64-linux-gnu/libpthread.so.0+0x000000000001138f
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   /lib/x86_64-linux-gnu/libc.so.6+0x0000000000035427
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   /lib/x86_64-linux-gnu/libc.so.6+0x0000000000037029
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   /lib/x86_64-linux-gnu/libc.so.6+0x000000000002dbd6
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   /lib/x86_64-linux-gnu/libc.so.6+0x000000000002dc81
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   0x0000000000f37662
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   0x0000000000534c6a
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   /lib/x86_64-linux-gnu/libc.so.6+0x000000000002082f
May 03 09:11:59 vladimir-ubuntu scylla[34966]:   0x00000000005a1a98
```

If the address is correct (no accidental backtick), everything loads and starts fine.

The instructions for manually installing Scylla on Ubuntu 16.04 say:

```
Configure Scylla

Configure the /etc/scylla/scylla.yaml file with the following parameters:
```
Item | Content
-- | --
cluster_name | Name of the cluster, all the nodes in the cluster must have the same name
seeds | Seed nodes are used during startup to bootstrap the gossip process and join the cluster
listen_address | IP address that the Scylla use to connect to other Scylla nodes in the cluster
rpc_address | IP address of interface for client connections (Thrift, CQL)
```
More information regarding the scylla.yaml file.
```

So it is expected that the user configures this parameter manually. It is not unusual to expect this kind of a typo (I've hit the backtick key while reaching for the "Esc" key in Vim).

**Comment by alekseys** [comment]
Same happens in CentOS 7.6. Also, when coredump is enabled, the problem is farther exaggerated by a long time and complete resource hog (even PS command stops producing any output) while the coredump is generated every time it restarts (which is repeated, so you can barely do anything but restart the machine).

**Comment by jwnx** [comment]
Reproduced on `3.0.5-0.20190414.403f66eca` Ubuntu 16.04 LTS.
Fixed on master (`7a0c6cd58`)

**Comment by slivne** [comment]
since it fixed on master closing
